test/hotspot/gtest/riscv/test_assembler_riscv.cpp

```diff
@@ -322,151 +326,374 @@ TEST_VM(RiscV, cmpxchg_int16_lr_sc) {
   UseZacas = zacas;
 }
 
+TEST_VM(RiscV, cmpxchg_int16_maybe_zacas) {
+  if (UseZacas) {
+    run_narrow_cmpxchg_tests<int16_t, Assembler::int16>();
+  }
+}
+
 TEST_VM(RiscV, cmpxchg_int8_lr_sc) {
   bool zacas = UseZacas;
   UseZacas = false;
   run_narrow_cmpxchg_tests<int8_t, Assembler::int8>();
   UseZacas = zacas;
 }
 
-TEST_VM(RiscV, cmpxchg_int16_maybe_zacas) {
+TEST_VM(RiscV, cmpxchg_int8_maybe_zacas) {
   if (UseZacas) {
-    run_narrow_cmpxchg_tests<int16_t, Assembler::int16>();
+    run_narrow_cmpxchg_tests<int8_t, Assembler::int8>();
   }
 }
 
-TEST_VM(RiscV, cmpxchg_int8_maybe_zacas) {
+template <typename TESTSIZE>
+TESTSIZE next_count(TESTSIZE now, TESTSIZE add) {
+  if ((std::numeric_limits<TESTSIZE>::max() - add) >= now) {
+    return now + add;
+  }
+  TESTSIZE diff = std::numeric_limits<TESTSIZE>::max() - now;
+  add -= diff;
+  return std::numeric_limits<TESTSIZE>::min() + add;
+}
+
+constexpr int64_t PAR_IT_END       = 10000;
+constexpr int64_t NUMBER_THREADS   = 4;
+constexpr int64_t TOTAL_ITERATIONS = NUMBER_THREADS * PAR_IT_END;
+
+template <typename TESTSIZE>
+constexpr TESTSIZE result_count() {
+  if (std::numeric_limits<TESTSIZE>::max() > (std::numeric_limits<TESTSIZE>::min() + TOTAL_ITERATIONS)) {
+    return std::numeric_limits<TESTSIZE>::min() + TOTAL_ITERATIONS;
+  }
+  int64_t range = std::numeric_limits<TESTSIZE>::max();
+  range -= std::numeric_limits<TESTSIZE>::min();
+  int64_t rest = TOTAL_ITERATIONS % range;
+  return std::numeric_limits<TESTSIZE>::min() + rest;
+}
+
+template<>
+constexpr int64_t result_count<int64_t>() {
+    return std::numeric_limits<int64_t>::min() + TOTAL_ITERATIONS;
```

Comment:
There is an indentation issue for this line. Maybe we can specialize `int32_t` as well?

Reply:
I wanted to cover when adding uint32.
Used enbaled if instead and added uint32.